### PR TITLE
STIX bundle optimizations

### PR DESCRIPTION
**Pull request by pcoccoli**
1. Download and save bundle files to avoid multiple downloads
2. Pre-compile STIX pattern before matching

### CODE REVIEW

File: src/kestrel_datasource_stixbundle/interface.py [hunk 1]
```diff
@@ -43,41 +48,78 @@ def query(uri, pattern, session_id=None, config=None):
...
         data_paths = data_paths.split(",")
         pattern = fixup_pattern(pattern)
+        compiled_pattern = Pattern(pattern)
 
         ingestdir = _make_query_dir(uri)
         bundles = []
         for i, data_path in enumerate(data_paths):
             data_path_striped = "".join(filter(str.isalnum, data_path))
-            ingestfile = ingestdir / f"{i}_{data_path_striped}.json"
+            bundlefile = ingestdir / f"{data_path_striped}.json"
```
Comment (subbyte) [inline src/kestrel_datasource_stixbundle/interface.py:57]: Each time `query()` is called, there should be a random `ingestdir`, which may make caching not friendly. Maybe we'd like to put all downloaded bundles not in `ingestdir/...` but at a fixed name dir, e.g., like `stixbundles` under the runtime dir. We may need to append the modified time to the file name explicitly (download time if that not exist) so the same name json from different sites will not collide.

Comment (pcoccoli) [reply src/kestrel_datasource_stixbundle/interface.py:57]: I think that's a further optimization that we could (and probably should) do, but I don't have time right now.

Comment (subbyte) [reply src/kestrel_datasource_stixbundle/interface.py:57]: I think my concern may not exist given we are using `uuid5()` to generate `ingestdir`. Please ignore it.

**Review by subbyte — APPROVED** [review]
smart idea to check caching with modified time. Looks good!

**Pull request merged**